Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: musl atoll() doesn't return ERANGE in case of overflow
----

--- a/gdal/port/cpl_conv.cpp
+++ b/gdal/port/cpl_conv.cpp
@@ -1006,8 +1006,6 @@ GIntBig CPLAtoGIntBig( const char *pszSt
 #endif
 }

-#if defined(__MINGW32__) || defined(__sun__)
-
 // mingw atoll() doesn't return ERANGE in case of overflow
 static int CPLAtoGIntBigExHasOverflow(const char* pszString, GIntBig nVal)
 {
@@ -1030,8 +1028,6 @@ static int CPLAtoGIntBigExHasOverflow(co
     return strcmp(szBuffer, pszString) != 0;
 }

-#endif
-
 /************************************************************************/
 /*                          CPLAtoGIntBigEx()                           */
 /************************************************************************/
@@ -1058,9 +1054,7 @@ GIntBig CPLAtoGIntBigEx( const char* psz
     GIntBig nVal = atol(pszString);
 #endif
     if( errno == ERANGE
-#if defined(__MINGW32__) || defined(__sun__)
         || CPLAtoGIntBigExHasOverflow(pszString, nVal)
-#endif
         )
     {
         if( pbOverflow )
